src/settings/mod.rs

```diff
@@ -22,7 +22,8 @@ use validate;
 mod test;
 
 macro_rules! deserialize_and_validate {
-    ($(($type:ident, $validator:ident, $expected:expr)),+) => ($(
+    ($(#[$docs:meta] ($type:ident, $validator:ident, $expected:expr)),+) => ($(
+        #[$docs]
```

Comment:
Cool stuff, didn't know you had to do this in order to add comments.

Reply:
> didn't know you had to do this in order to add comments

No worries, it took me a while to figure out how to do it! 😄